scRNA-seq-advanced/01-reading_filtering_CellRanger.Rmd

```diff
@@ -0,0 +1,484 @@
+In this case, we will use our `mito_genes` vector to define a subset called `mito`.
+The `mito` name is important in that it is the name that will be expected by a later function.
+(We could define more subsets, but for now this one will do.)
+
+```{r per cell QC}
+filtered_sce <- scuttle::addPerCellQC(filtered_sce,
+                                      subsets = list(mito = mito_genes))
+```
+
+Now we can look at the colData to see what was added:
+
+```{r view coldata stats}
+head(colData(filtered_sce))
+```
+
+We can also plot some of these statistics, here using the `plotMetrics()` function from the `miQC` package to plots the percent of reads that are mitochondrial (the `subsets_mito_percent` column) against the number of unique genes detected (the `detected` column) for each cell.
+
+```{r miQC plotMetrics}
+# use miQC::plotMetrics()
+miQC::plotMetrics(filtered_sce) + theme_bw()
+```
+
+We can see that there is a range of mitochondrial percentages, and it does also seem that genes with high percentages of mitochondrial cells don't seem to detect very many unique genes.
+
+One option is to define a cutoff for the mitochondrial percentage above which we call a cell compromised and exclude it from further analysis. 
+However, deciding on that cutoff can be a bit fraught, as the expected percentage of mitochondrial reads can vary depending on the cell type and library preparation methods. 
+So it might be nice to have a method to determine that cutoff from the data itself.
+
+### Filtering compromised cells
+
+Determining mitochondrial cutoffs is exactly what the `miQC` package does ([Hippen _et al._ 2021] (https://doi.org/10.1371/journal.pcbi.1009290))!
+In truth, it does something possibly even a bit better: it fits a mixture model to the data that consists of distributions of healthy cells and compromised cells.
+Then we can calculate whether each cell is more likely to belong to the healthy or compromised distribution.
+We can then exclude the cells that are more likely to be compromised. 
+
+To use miQC, we first fit a model to the data in our SCE object:
+
+```{r miQC model}
+miqc_model <- miQC::mixtureModel(filtered_sce) 
+```
+
+Now we can plot the model results to see how it corresponds to our data.
+We should expect to see two fit lines: 
+One will correspond the the "healthy" cells and should show little to no relationship between the number of unique genes detected and the mitochondrial percentage.
+By contrast, the line that corresponds to compromised cells will show a negative relationship between the number of unique genes detected and the mitochondrial percentage.
+
+This plot will also show, for each cell, the posterior probability that the cell is derived from the compromised distribution; a higher score indicates that a cell is more likely to be compromised.
+
+It is also critical to note that this model can _and does_ fail at times. 
+Plotting the results as we have done here is not a step to skip. 
```

Comment:
A bit more "aggressive"  if you want... 
```suggestion
Do not skip the step plotting the results!
```

Reply:
I got aggressive in the next sentence, I thought!